### PR TITLE
[imagemanager] Remove state storage

### PR DESCRIPTION
**Pull request by MykolaSolyanko**
Signed-off-by: Mykola Solianko <mykola_solianko@epam.com>

### CODE REVIEW

**Review by Eabramov84 — APPROVED** [review]

File: imagemanager/imagemanager.go [hunk 1]
```diff
@@ -838,6 +845,10 @@ func (imagemanager *Imagemanager) removeOutdatedService(serviceID string) error
 		}
 	}
 
+	if err := imagemanager.state.Remove(serviceID); err != nil {
```
Comment (Eabramov84) [inline imagemanager/imagemanager.go:848]: if we can't remove service storage, we have to try remove state

File: imagemanager/imagemanager.go [hunk 2]
```diff
@@ -81,13 +81,19 @@ type CryptoContext interface {
 	CreateSignContext() (fcrypt.SignContextInterface, error)
 }
 
+// StateStorage provides API to remove state storage.
```
Comment (al1img) [inline imagemanager/imagemanager.go:84]: Comment doesn't correspond to interface. Name it `StateRemover`, or `StateStorageRemove`. Good article how to name interfaces in go: https://medium.com/@dotronglong/interface-naming-convention-in-golang-f53d9f471593

Comment (MykolaSolyanko) [reply imagemanager/imagemanager.go]: Done

File: imagemanager/imagemanager.go (hunk 1 above)
Comment (MykolaSolyanko) [reply imagemanager/imagemanager.go]: Done

File: imagemanager/imagemanager.go [hunk 3]
```diff
@@ -81,13 +81,19 @@ type CryptoContext interface {
 	CreateSignContext() (fcrypt.SignContextInterface, error)
 }
 
+// StateStorageRemove provides API to remove state storage.
+type StateStorageRemove interface {
```
Comment (al1img) [inline imagemanager/imagemanager.go:85]: Remover

Comment (MykolaSolyanko) [reply imagemanager/imagemanager.go]: Done

**Review by al1img — APPROVED** [review]

**Pull request merged**